update_cache.py

```diff
@@ -0,0 +1,81 @@
+"""
+Run within a kick_tuxsuite_foo job.
```

Comment:
Right.

My thinking is that there is no way in hell that two of these `kick...` jobs will finish at the exact same time. Even if they did GitHub API should queue the writes. Moreover, I have implemented logic to ensure that we cannot change a `fail` to a `pass`.

With this, all the `kick...` jobs must write exactly `pass` to our cache otherwise the status will stay as `fail` or other failure status.

I am trying to be extremely conservative of when the cache steps in to skip a build and when it doesn't.

Reply:
Ah, I missed `allow_fail_to_pass` in `update_repository_variable()`, I remember us talking about that now. I think that is a reasonable way to handle this.